### PR TITLE
PLANET-4070 Convert auto-generated shortcakes to Gutenberg

### PR DESCRIPTION
**Pull request by sagarsdeshmukh**
[JIRA 4070](https://jira.greenpeace.org/browse/PLANET-4070)

Convert autogenerated shortcake shortcodes into new Gutenberg blocks.

### CODE REVIEW

**Comment by pablocubico** [comment]
@sagarsdeshmukh I'm seeing this shortcake code on the bottom of a tag page:

`[shortcake_happy_point focus_image=”left top” /]`

This is in `/tag/renewables/`, is that expected? Otherwise, it looks good!

File: single.php [hunk 1]
```diff
@@ -22,10 +22,10 @@
 $context['post'] = $post;
 
 // Strip Take Action Boxout block from the post content to add outside the normal block container.
-if ( false !== strpos( $post->post_content, '[shortcake_take_action_boxout' ) ) {
+if ( false !== strpos( $post->post_content, '<!-- wp:planet4-blocks/take-action-boxout' ) ) {
 
-	$take_action_boxout_block_start  = strpos( $post->post_content, '[shortcake_take_action_boxout' );
-	$take_action_boxout_block_end    = strpos( $post->post_content, ']', $take_action_boxout_block_start );
+	$take_action_boxout_block_start  = strpos( $post->post_content, '<!-- wp:planet4-blocks/take-action-boxout' );
+	$take_action_boxout_block_end    = strpos( $post->post_content, '/-->', $take_action_boxout_block_start );
```
Comment (dpivo) [inline single.php]: You will need to add 3 to this value to close the comment properly

**Comment by sagarsdeshmukh** [comment]
> @sagarsdeshmukh I'm seeing this shortcake code on the bottom of a tag page:
> 
> `[shortcake_happy_point focus_image=”left top” /]`
> 
> This is in `/tag/renewables/`, is that expected? Otherwise, it looks good!

Is it on Gutenberg instance? could you please share the link, if available

**Comment by pablocubico** [comment]
> > @sagarsdeshmukh I'm seeing this shortcake code on the bottom of a tag page:
> > `[shortcake_happy_point focus_image=”left top” /]`
> > This is in `/tag/renewables/`, is that expected? Otherwise, it looks good!
> 
> Is it on Gutenberg instance? could you please share the link, if available

This was happening locally, but if you (and Dylan) don't see it, maybe it was an error on my end.

File: classes/class-p4-taxonomy-campaign.php [hunk 2]
```diff
@@ -49,15 +49,17 @@ public function add_block( $block_name, $data ) {
 
 			if ( $block_name && $data ) {
 
-				$shortcode = '[shortcake_' . $block_name;
+				$blockcode = '<!-- wp:planet4-blocks/' . $block_name . ' {';
 				if ( $data ) {
+					$param_array = [];
```
Comment (kirdia) [inline classes/class-p4-taxonomy-campaign.php]: This is not readable. No need to concatenate parameters.
Pass the parameters here as an assosiative array with the correct type for each value, json encode the array and build the gutenberg block string

File: single.php [hunk 3]
```diff
@@ -65,7 +65,7 @@
 
 // Build the shortcode for articles block.
 if ( 'yes' === $post->include_articles ) {
-	$post->articles = "[shortcake_articles exclude_post_id='" . $post->ID . "' /]";
+	$post->articles = '<!-- wp:planet4-blocks/articles {"exclude_post_id":"' . $post->ID . '"} /-->';
```
Comment (kirdia) [inline single.php]: this should be integer

File: classes/class-p4-campaigns.php [hunk 4]
```diff
@@ -250,13 +250,13 @@ public function save_taxonomy_meta( $term_id ) {
 				$tag_data = get_term( $term_id );
 
 				if ( $tag_data instanceof \WP_Term ) {
-					$post_content = '[shortcake_newcovers cover_type="1" tags="' . $term_id . '" covers_view="0" title="' . __( 'Things you can do', 'planet4-master-theme' ) . '"  description="' . __( 'We want you to take action because together we\'re strong.', 'planet4-master-theme' ) . '" /]
+					$post_content = '<!-- wp:planet4-blocks/covers {"title":"' . __( 'Things you can do', 'planet4-master-theme' ) . '","description":"' . __( 'We want you to take action because together we\'re strong.', 'planet4-master-theme' ) . '","tags":[' . $term_id . '],"posts":[],"post_types":[],"cover_type":"1","covers_view":"0"} /-->
```
Comment (kirdia) [inline classes/class-p4-campaigns.php]: same here, please populated an array with the attributes and then generate the json body for the block

File: classes/class-p4-taxonomy-campaign.php (hunk 2 above)
Comment (sagarsdeshmukh) [reply classes/class-p4-taxonomy-campaign.php]: fixed.

**Comment by sagarsdeshmukh** [comment]
@kirdia - I updated the PR with code review fixes. please have a look.

File: classes/class-p4-campaigns.php [hunk 5]
```diff
@@ -250,13 +250,44 @@ public function save_taxonomy_meta( $term_id ) {
...
+						'tags'        => [ $term_id ],
+						'posts'       => [],
+						'post_types'  => [ 62 ],
+						'cover_type'  => '3',
+						'covers_view' => '0',
+					];
+
+					$happypoint_block_attributes = [
+						'mailing_list_iframe' => true,
+						'opacity'             => $happypoint_bg_opacity ?? '30',
```
Comment (kirdia) [inline classes/class-p4-campaigns.php]: should this be an integer?

File: classes/class-p4-campaigns.php [hunk 6]
```diff
@@ -250,13 +250,44 @@ public function save_taxonomy_meta( $term_id ) {
...
+						'posts'       => [],
+						'post_types'  => [],
+						'cover_type'  => '1',
+						'covers_view' => '0',
+					];
 
-[shortcake_newcovers cover_type="3" title="' . __( 'Publications', 'planet4-blocks' ) . '" tags="' . $term_id . '" post_types="62" covers_view="0" /]
+					$articles_block_attributes = [
+						'tags'              => [ $term_id ],
+						'ignore_categories' => 'false',
```
Comment (kirdia) [inline classes/class-p4-campaigns.php]: this should be a boolean

File: tag.php [hunk 7]
```diff
@@ -93,16 +92,18 @@
 			}
 		} else {
 			// If none is selected, then display Publications by default (for backwards compatibility).
-			$cfc_args['p4_page_type_publication'] = 'true';
+			$cfc_args['p4_page_type_publication'] = '"true"';
```
Comment (kirdia) [inline tag.php]: this can be removeds, it is an old attribute

**Pull request merged**